src/dataflow/core/linearize.py

```diff
@@ -12,12 +12,24 @@
     Lispress,
     lispress_to_program,
     program_to_lispress,
+    render_compact,
 )
 from dataflow.core.program import Program
 from dataflow.core.program_utils import Idx, OpType
 from dataflow.core.sexp import Sexp
 
 
+def to_canonical_form(tokenized_lispress: str) -> str:
+    """Returns canonical form of a tokenized lispress.
+
+    The canonical form is un-tokenized and compact; it also sorts named arguments in alphabetical order.
+    """
+    lispress = seq_to_lispress(tokenized_lispress.split(" "))
```

Comment:
this is where the leading and trailing spaces are getting trimmed, right?

Reply:
Yes, I think so.
So you will need to change the leaderboard code accordingly.